### PR TITLE
Simple 1D FCT advection test

### LINKED ISSUE #878: FCT: Add simple integration/unit test
**Is your feature request related to a problem? Please describe.**
In #872 we added spatial convergence unit tests for the simple vanilla FCT. 
It would be good to also add a simple integration/unit test, like an advection equation of a square wave (pulse), so that we can see the isolated effects of using only low-order upwinding, high-order upwinding or a combination of the two (like in FCT).

**Please link to any related issues/PRs.**

This issue is incremental work towards more complicated cases #851 

**Describe the solution you'd like**
Add a simple 1D column advection of a square wave (pulse) example.

### PR DESCRIPTION
**Pull request by valeriabarra**
This will close #878 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by valeriabarra** [comment]
bors try

**Comment by valeriabarra** [comment]
This example is very nice to play with, for people who want to learn more about dissipation and dispersion :) 

We can see how the first-order upwinding is more dissipative (i.e., it has no overshoots/undershoots), but very dispersive (i.e., the pulse wave has shifted laterally significantly) :
exact_and_computed_advected_square_wave_C_0

Vs the third-order upwinding, which is less dissipative compared to the first-order one (it shows overshoots/undershoots) but way less dispersive:

exact_and_computed_advected_square_wave_C_1

And the simple FCT that combines the two in a simple linear combination (average) fashion, where we see the lower peak values in the overshoots/undershoots compared to the third-order-upwinding alone, and limited dispersion, compared to the first-order-upwinding alone:
exact_and_computed_advected_square_wave_C_05

File: examples/column/fct_advection.jl [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+w = Geometry.WVector.(speed .* ones(FT, fs))
+
+# Define a pulse wave or square wave
+pulse(z, t, z₀, zₕ, z₁) = abs(z.z - speed * t) ≤ zₕ ? z₁ : z₀
+
+# Initial condition
+y0 = pulse.(zc, 0.0, z₀, zₕ, z₁)
+
+function f!(dydt, y, parameters, t)
+
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:70]:
```suggestion
    (; w, C) = parameters
    FT = Spaces.undertype(y)
```

File: examples/column/fct_advection.jl [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    )
+    divf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0.0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0.0))),
+    )
+
+    corrected_antidiff_flux = similar(dydt)
+    @. corrected_antidiff_flux =
+        divf2c(C * (third_order_fluxᶠ(w, y) - first_order_fluxᶠ(w, y)))
+    @. dydt = -(divf2c.(first_order_fluxᶠ(w, y)) + corrected_antidiff_flux)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:87]:
```suggestion
    @. dydt = -(divf2c(first_order_fluxᶠ(w, y)) + corrected_antidiff_flux)
```

File: examples/column/fct_advection.jl [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    return dydt
+end
+
+# Solve the ODE operator
+Δt = 0.001
+dydt = copy(y0)
+parameters = (;)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:95]:
```suggestion
parameters = (; w, C)
```

File: examples/column/fct_advection.jl [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+# Unitary, constant advective velocity
+w = Geometry.WVector.(speed .* ones(FT, fs))
+
+# Define a pulse wave or square wave
+pulse(z, t, z₀, zₕ, z₁) = abs(z.z - speed * t) ≤ zₕ ? z₁ : z₀
+
+# Initial condition
+y0 = pulse.(zc, 0.0, z₀, zₕ, z₁)
+
+function f!(dydt, y, parameters, t)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:69]: Let's move this function up, before `FT = Float64`, to ensure that we've removed all closures.

File: examples/column/fct_advection.jl [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+analytical_result = pulse.(zc, t₁, z₀, zₕ, z₁)
+err = norm(computed_result .- analytical_result)
+
+@test err ≤ 0.075
+
+plot(sol.u[end])
+Plots.png(
+    Plots.plot!(analytical_result, title = "FCT (C=0.5)"),
+    joinpath(path, "exact_and_computed_advected_square_wave_C_05.png"),
+)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:118]: Should we include what the solution looks like, too?

File: examples/column/fct_advection.jl (hunk 1 above)
Comment (charleskawczynski) [reply examples/column/fct_advection.jl:70]: This, together with the changes below, should make this example run significantly faster.

Comment (valeriabarra) [reply examples/column/fct_advection.jl:70]: Thanks Charlie. Just one clarification. We declare FT at the top of the file here. You're saying we should get it from the field instead?

Comment (valeriabarra) [reply examples/column/fct_advection.jl:70]: Oh, I think I see what you meant now!

File: examples/column/fct_advection.jl (hunk 5 above)
Comment (valeriabarra) [reply examples/column/fct_advection.jl:118]: Yes, I used `plot!` to superimpose both the analytic solution and the computed one (it's visible in the CI buildkite artifact)

File: examples/column/fct_advection.jl (hunk 1 above)
Comment (valeriabarra) [reply examples/column/fct_advection.jl:70]: And just for future reference, the syntax for the suggestion on the second line is:
```
FT = Spaces.undertype(axes(y))
```

**Comment by valeriabarra** [comment]
bors try

File: examples/column/fct_advection.jl (hunk 1 above)
Comment (charleskawczynski) [reply examples/column/fct_advection.jl]: Yes, grabbing `FT` from global scope above is a performance pitfall. Grabbing it from within the scope of `f!` fixes this issue, and `Spaces.undertype(y)` is one way to do this.

Comment (valeriabarra) [reply examples/column/fct_advection.jl]: Thanks. I applied all suggestions. Waiting for approval and buildkite before squashing+merging.

File: examples/column/fct_advection.jl [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    # link figure in logs if we are running on CI
+    if get(ENV, "BUILDKITE", "") == "true"
+        artifact_url = "artifact://$figpath"
+        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+    end
+end
+
+function f!(dydt, y, parameters, t)
+
+    (; w, C) = parameters
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:40]:
```suggestion
    (; w, C, corrected_antidiff_flux) = parameters
```

File: examples/column/fct_advection.jl [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    third_order_fluxᶠ = Operators.Upwind3rdOrderBiasedProductC2F(
+        bottom = Operators.ThirdOrderOneSided(),
+        top = Operators.ThirdOrderOneSided(),
+    )
+    divf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0.0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0.0))),
+    )
+
+    corrected_antidiff_flux = similar(dydt)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:56]:
```suggestion
```

File: examples/column/fct_advection.jl [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+# Define a pulse wave or square wave
+pulse(z, t, z₀, zₕ, z₁) = abs(z.z - speed * t) ≤ zₕ ? z₁ : z₀
+
+# Initial condition
+y0 = pulse.(zc, 0.0, z₀, zₕ, z₁)
+
+# Solve the ODE operator
+Δt = 0.001
+dydt = copy(y0)
+parameters = (; w, C)
```
Comment (charleskawczynski) [inline examples/column/fct_advection.jl:98]:
```suggestion
corrected_antidiff_flux = similar(y0)
parameters = (; w, C, corrected_antidiff_flux)
```

File: examples/column/fct_advection.jl (hunk 5 above)
Comment (charleskawczynski) [reply examples/column/fct_advection.jl:121]: Perfect, thanks!

**Review by charleskawczynski — APPROVED** [review]
I added one more suggestion, good to squash/merge after that, thanks!

**Comment by valeriabarra** [comment]
bors try

**Pull request merged**